src/irclog2html/logs2html.py

```diff
@@ -239,7 +239,13 @@ def process(dir, options):
             extra_args += ['--output-file', options.output_dir]
             out_dir = options.output_dir
         else:
-            raise Error('Argument specified in --output-file is not a directory')
+            if os.path.isfile(options.output_dir):
+                raise Error("%s is a file" % options.output_dir)
+            else:
+                try:
+                    os.makedirs(options.output_dir)
+                except Error as e:
+                    sys.exit("Failed to recursively create directory %s: %s" % (options.output_dir, e))
```

Comment:
Better make this `raise Error("Failed to create directory %s: %s" % (...))`.

Reply:
TBH I'm not sure why I introduced an `Error` class meaning "terminate program and print this error message to the user", when, as you've already noticed, SystemExit does exactly the same.

Still, consistency is good.